tests/test_raw_node.rs

```diff
@@ -141,6 +141,56 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_read_index ensures that RawNode.read_index sends the MsgReadIndex message
+// to the underlying raft. It also ensures that ReadState can be read out.
+#[test]
+fn test_raw_node_read_index() {
+    let wrequest_ctx = b"somedata".to_vec();
+    let wrs = vec![
+        ReadState {index: 2u64, request_ctx: wrequest_ctx.clone()},
```

Comment:
2 is the commit index for a new leader. 1 entry of member change for startup, 1 entry of no-op after becoming leader.